.github/workflows/ci.yml

```diff
@@ -0,0 +1,238 @@
+name: CI
+
+# Trigger on pull_request or on push to master
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: install tools
+      run: |
+        sudo pip3 install flake8==3.7.8
+        sudo apt-get install clang-format clang-tidy
+        echo "::add-path::/usr/lib/llvm-8/bin"
+    - name: flake8
+      run: flake8
+    - run: ./scripts/clang-format-diff.sh
+    - run: ./scripts/clang-tidy-diff.sh
+
+  build:
+    name: build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+
+    - name: install ninja (linux)
+      run: sudo apt-get install ninja-build
+      if: matrix.os == 'ubuntu-latest'
+
+    - name: install ninja (osx)
+      run: brew install ninja
+      if: matrix.os == 'macos-latest'
+
+    - name: install ninja (win)
+      run: choco install ninja
+      if: matrix.os == 'windows-latest'
+
+    - name: mkdir
+      run: mkdir -p out
```

Comment:
how does this work on windows?

Reply:
I think the actions run under the git-bash shell by default which is some kind MINGW/MSYS version of bash.

Reply:
Sorry, late to the party. If you type this into cmd, you'll get two directories `-p` and `out`. So this might be what's happening here.

Reply:
Just learned: Under PowerShell `mkdir -p out` is an alias of `mkdir -Path out`, which apparently does what `mkdir -p` does. So, all good.